### ISSUE TITLE
Error with required type: password in rancher compose

### ISSUE THREAD

**Issue by mccricardo** [rancher/rancher#5793]
**Rancher Version:**
v1.2.0-pre2

**Docker Version:**
 1.12.1

**OS and where are the hosts located? (cloud, bare metal, etc):**
Ubuntu 16.04

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
Create configuration where you have **type:password** in **rancher.compose.yml**. Example:

``` yml
.catalog:
  version: 1.0.0
  questions:
    - variable: github_client_secret
      label: GitHub client secret
      required: true
      type: password
    - variable: drone_secret
      label: secret used by Drone agent to communicate with Server
      required: true
      type: password
```

When adding that value via the interface, the added value is not accepted unless the button **'Generate'** is used. Example:

![rancher](https://cloud.githubusercontent.com/assets/1016505/17937637/b1d43b72-6a1a-11e6-98cb-c30d826a38b7.png)

Curiously the other **type:password** (the one right below in the example picture) works fine.

**Results:** 
Unable to launch stack unless **'Generate'** button is used to fill in the field.

**Expected:**
All **type:password** fields should allow being introduced.
## Update

Digging a little deeper, I've found that any required password fields in the first row are not accepted. Example **rancher-compose.yml**:

``` yml
.catalog:
  version: 1.0.0
  questions:
    - variable: secret_var_1
      label: Secret Var 1
      required: true
      type: password
    - variable: secret_var_2
      label: Secret Var 2
      required: true
      type: password
    - variable: secret_var_3
      label: Secret Var 3
      required: true
      type: password
```

Check pictures below.

![rancher-1](https://cloud.githubusercontent.com/assets/1016505/17939016/783a2e02-6a20-11e6-8321-f517058d1dc4.png)
![rancher-2](https://cloud.githubusercontent.com/assets/1016505/17939017/783ce2c8-6a20-11e6-8066-2aa078db59f1.png)

As you can see in the first picture, when all fields are empty, Rancher behaves as expected. In the second picture we can see that, although, all fields are filled in, only the bottom one is accepted.

**Comment by vincent99** [comment]
This appears to have been because they all get the same `id="password"` in the HTML, which makes only the last password field work (not one on the first row, etc)

Anyway this was fixed right after 1.2.0-pre3 in https://github.com/rancher/ui/commit/52506581ba9f80e99d3cea7e3a7dc7156492858a#diff-b8a249cd297ed3c367887a8adc9059e9L1

**Comment by tfiduccia** [comment]
Version - master 11/09
Verified fixed
